Yoast/wordpress-seo
# Code coverage presentation tests

## Summary


This PR can be summarized in the following changelog entry:

* Adds tests for the classes in the `presentation` folder to get to 100% coverage for these files.

## Relevant technical choices:

* Added a mock class for the `Abstract_Presentation` to be able to mock `is_prototype`.
* Simplified the `generate_open_graph_article_publisher` method to a wrapper, since its method call to `context->open_graph_publisher` already defaulted to the empty string.
  * Removed existing tests for this method and added a `@codeCoverageIgnore` tag.

## Test instructions

This PR can be tested by following these steps:

**Note**: Running code coverage of the entire unit tests does not seem to work entirely for `abstract_presentation.php`. However, running these tests separately **does** give a coverage of 100% for this file.

* Run the tests in `tests/presentations/abstract_presentation_test.php` with code coverage enabled.
* See that the coverage of `src/presentations/abstract_presentation.php` is 100%.
* Run all unit tests with code coverage enabled.
* See that the coverage of the `presentation` folder is 100%, **except** for `abstract_presentation.php`.

## UI changes

## Documentation

## Quality assurance


Fixes #
